samples/wavlib/src/main/cpp/io/stream/InputStream.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _IO_STREAM_INPUTSTREAM_H_
+#define _IO_STREAM_INPUTSTREAM_H_
+
+#include <cstdint>
+
+namespace wavlib {
+
+/**
+ * An interface declaration for a stream of bytes. Concrete implements for File and Memory Buffers
+ */
+class InputStream {

```

Comment:
Couldn't this be provided by the std IO library? https://en.cppreference.com/w/cpp/io

Reply:
Something to look into in a later CL.